Neos.ContentRepository.Core/Classes/NodeType/NodeType.php

```diff
@@ -413,18 +413,11 @@ public function getProperties(): array
      *
      * @param string $propertyName Name of the property
      */
-    public function getPropertyType(string $propertyName): string
+    public function getPropertyType(string $propertyName): ?string
     {
         $this->initialize();
 
-        if (
-            !isset($this->fullConfiguration['properties'])
-            || !isset($this->fullConfiguration['properties'][$propertyName])
-            || !isset($this->fullConfiguration['properties'][$propertyName]['type'])
-        ) {
-            return 'string';
-        }
-        return $this->fullConfiguration['properties'][$propertyName]['type'];
+        return $this->fullConfiguration['properties'][$propertyName]['type'] ?? null;
```

Comment:
fallback to "string" if `$propertyName` actually exists, but has no type.

case `$propertyName` doesnt exist:
-> throw exception instead of nullable -> and introduce `hasProperty`

Reply:
> fallback to "string" if $propertyName actually exists, but has no type.

That fallback should be moved to the _creation_ side of things IMO

Reply:
yes but i would take care of this another time.